ArmPkg/Library/ArmFfaLib/ArmFfaRxTxMap.c

```diff
@@ -0,0 +1,256 @@
+/** @file
+  Arm Ffa library common code.
+
+  Copyright (c) 2024, Arm Limited. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+   @par Glossary:
+     - FF-A - Firmware Framework for Arm A-profile
+
+   @par Reference(s):
+     - Arm Firmware Framework for Arm A-Profile [https://developer.arm.com/documentation/den0077/latest]
+
+**/
+#include <Uefi.h>
+#include <Pi/PiMultiPhase.h>
+
+#include <Library/ArmLib.h>
+#include <Library/ArmFfaLib.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PcdLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+
+#include <IndustryStandard/ArmFfaSvc.h>
+
+#include "ArmFfaCommon.h"
+#include "ArmFfaRxTxMap.h"
+
+/**
+  Get mapped Rx/Tx buffers.
+
+  @param [out]   TxBuffer         Address of TxBuffer
+  @param [out]   TxBufferSize     Size of TxBuffer
+  @param [out]   RxBuffer         Address of RxBuffer
+  @param [out]   RxBufferSize     Size of RxBuffer
+
+  @retval EFI_SUCCESS
+  @retval Others             Error.
+
+**/
+EFI_STATUS
+EFIAPI
+ArmFfaLibGetRxTxBuffers (
+  OUT VOID    **TxBuffer,
+  OUT UINT64  *TxBufferSize,
+  OUT VOID    **RxBuffer,
+  OUT UINT64  *RxBufferSize
+  )
+{
+  if ((PcdGet64 (PcdFfaTxBuffer) == 0x00) || (PcdGet64 (PcdFfaRxBuffer) == 0x00)) {

```

Comment:
Can we use a local variable to store the value returned by PcdGet64 (PcdFfaTxBuffer) and PcdGet64 (PcdFfaRxBuffer)?

Reply:
Same for PcdGet64 (PcdFfaTxRxPageCount) * EFI_PAGE_SIZE;

Reply:
> Can we use a local variable to store the value returned by PcdGet64 (PcdFfaTxBuffer) and PcdGet64 (PcdFfaRxBuffer)?

Done.

Reply:
> Same for PcdGet64 (PcdFfaTxRxPageCount) * EFI_PAGE_SIZE;

In case of Size, Because it only read when Size arguments passed, I think It doesn't need to read with local variable.